tests/std/tests/GH_003131_tuple_opeq_overload_and/test.cpp

```diff
@@ -0,0 +1,57 @@
+#include <cassert>
+#include <tuple>
+
+using namespace std;
+
+struct WeirdBoolBinOp {
+    bool b;
+
+    operator bool() const {
+        return b;
+    }
+    WeirdBoolBinOp operator!() const {
+        return WeirdBoolBinOp{!b};
+    }
+
+    // EVILLLLL
+    friend bool operator&&(const WeirdBoolBinOp& lhs, bool rhs) {
+        return !lhs.b && rhs;
+    }
+    friend bool operator||(const WeirdBoolBinOp& lhs, bool rhs) {
+        return !lhs.b || rhs;
+    }
+};
+struct WeirdBoolNot {

```

Comment:
I think this is a good spot for a paragraph break:
```suggestion
};

struct WeirdBoolNot {
```